### DISCUSSION TITLE
Suggestion: How about setting og:image in the MDN document on a per-page basis?

### DISCUSSION THREAD

**Discussion by debiru** [mdn/mdn-community#472] [Platform]
# Background

I am a contributor who localizes Japanese. So, I do a daily tweet search to see what people's feedback is on MDN documents.

For example, I check this URL: https://twitter.com/search?q=url%3Ahttps%3A%2F%2Fdeveloper.mozilla.org%2F&f=live

Then I noticed a row of uninterested og:images.

![og:image of the MDN pages](https://developer.mozilla.org/mdn-social-share.cd6c4a5a.png)

The `og:image:alt` is `The MDN Web Docs logo, featuring a blue accent color, displayed on a solid black background.`

# Questions

- (1) Has MDN og:image been discussed in the past?
- (2) Is it technically possible to generate an og:image for each article (with the title of the article displayed)?

**Comment by bsmth** [comment]
Thanks a lot for creating this one. 

1: I don't know if there's been a discussion about this yet, but I think it would be really nice to have. This might be the discussion :)

2: Not speaking for engineering, but this is technically possible. Probably the og:image that you're used to seeing the most is for GitHub links (repos, issues etc.) which is dynamically created at the time the request is made. As far as I know, the common approach is to use lambdas/edge functions where you have a HTML & CSS template into which you inject the page title at request time. This gets rendered with a headless browser, a screenshot is taken and then sent back as the response.

It would be great to investigate it more, probably there's a lot of practicalities to consider, which data should be dynamic, which sections of the site could share common images, and how to cache responses given the high volume of requests that would be made using this method.

I'd love to hear other opinions on it!
